### ISSUE TITLE
Instance start failure due to inability to place VMM should be plumbed up to the instance create API response

### ISSUE THREAD

**Issue by askfongjojo** [oxidecomputer/omicron#5122]
When the control plane cannot find a sled with the required vcpu and memory to place a VMM, instance start would fail with a HTTP 507 InsufficientCapacity error. This error is returned to the API caller only during the instance start operation but not the instance create operation. Currently, an instance create request that successfully provisions the VM but fails to start it up still returns a HTTP 200 response, resulting in confusion to the user that the VM is ready to be used when it's not.

**Comment by askfongjojo** [comment]
Talked to the customer just now. They expressed that if the change causes the instance start operation to be a synchronous part of instance create, it is also not ideal. The current behavior is acceptable to them and they can simply adjust their workflow to do an extra check on instance state after provisioning.

**Comment by zephraph** [comment]
There's another thing to consider here: If the response was plumbed through the API (e.g. a 400 or 500 error code happened when the user was trying to create and start and instance but it failed to start) what would we want to do with the state of the system? If that meant we unwound the creation of the instance then the user would have to start all over with the creation. Doesn't seem like that would be a good user experience.

Ultimately creating and instance and starting an instance are two separate processes and I think the interaction that we have to day is the least bad version. What we eventually want to do is highlight to a user that an instance they created failed to start.

**Comment by askfongjojo** [comment]
> If that meant we unwound the creation of the instance then the user would have to start all over with the creation. Doesn't seem like that would be a good user experience.

Agreed. The intent of this request is to pass on the error message without unwinding the create request. But this may be an anti-pattern of the saga workflow (i.e. any event failure would normally trigger the unwind workflow), causing the implementation more difficult/awkward than expected. Maybe we can pass back a HTTP 207 in this case to indicate partial success along with an appropriate error message?
